src/python/io.cpp

```diff
@@ -15,6 +13,55 @@
 #include "awkward/python/io.h"
 
 namespace ak = awkward;
+namespace {
+  class NumpyBuffersContainer: public ak::BuffersContainer {
+  public:
+    py::dict container() {
+      return container_;
+    }
+
+    void
+      copy_buffer(const std::string& name, const void* source, int64_t num_bytes) override {
+        py::object pyarray = py::module::import("numpy").attr("empty")(num_bytes, "u1");
+        py::array_t<uint8_t> rawarray = pyarray.cast<py::array_t<uint8_t>>();
+        py::buffer_info rawinfo = rawarray.request();
+        std::memcpy(rawinfo.ptr, source, num_bytes);
+        container_[py::str(name)] = pyarray;
+      }
+
+    void
+      full_buffer(const std::string& name, int64_t length, int64_t value, const std::string& dtype) override {
+        py::object pyarray = py::module::import("numpy").attr("full")(py::int_(length), py::int_(value), py::str(dtype));
+        container_[py::str(name)] = pyarray;
+      }
+
+  private:
+    py::dict container_;
+  };
```

Comment:
This is a direct copy-paste of

https://github.com/scikit-hep/awkward-1.0/blob/6f622b4ba609e04ea8eb5416d1025990d9c9f4e6/src/python/content.cpp#L979-L1002

Can they be consolidated? Does io.cpp and content.cpp have a common .h file that this class can be defined in?

We don't want it to be a visibly exported symbol, so its (tiny) implementations should remain inline.

Reply:
Thanks, it's done. I'd prefer to have a separate PR for `json` vs `ndjson` separation. I'd like to check if this is already sufficient to unblock the `ArrayBuilder` pending PR. Thanks!

Reply:
That's much better—thanks!